File: help/pt_pt/exercicios_da_ficha_11/Ficha11_016/Ficha11_016.cpp
// Programador           -

#include <iostream>
#include <cstring>
#include <cmath>
using namespace std;

// Class definition
class Ficha11_016 {

    public:
        int main() {
            cout << "Altura   : ";
            int alt;
            cin >> alt;
            cout << "Caractér : ";
            string char;
            cin >> char;
            copa(alt , char ) ;
            return 0;
        }

        //::::::::::::::::::::::::::::::::::::::::::::::::::
        //User Defined Functions Code
        //::::::::::::::::::::::::::::::::::::::::::::::::::
        void linhaCopa( int sp, int chars, string txt ){
            int i  = 0 ;
            for( int i = 0; i < sp; i = i + 1 ) {
                cout << " ";
            }
            int i  = 0 ;
            for( int i = 0; i < chars; i = i + 1 ) {
                cout << txt ;
            }
            cout << "\n";
        }
        //::::::::::::::::::::::::::::::::::::::::::::::::::
        void copa( int alt, string txt ){
            int i  = 0 ;
            for( int i = 1; i < alt + 1; i = i + 1 ) {
                int sp  = alt - i ;
                int chrs1  = i * 2 - 1 ;
                linhaCopa(sp , chrs1 , txt ) ;
            }
        }
};// end of class Ficha11_016
// Execute main function
int main(){
    Ficha11_016 application; // create object
    application.main(); // execute object
    return 0;
}

//C++  - Automatic translated by Algorithmi 22.05
//(c) Ant0nio M@nso 	quarta, 2022 out 12 20:15

File: help/pt_pt/exercicios_da_ficha_11/Ficha11_016/Ficha11_016.cs
// Programador           -

using System.IO;
using System;
public class Ficha11_016 {

    //Main Function
    static void Main() {
        Console.Write("Altura   : " );
        int alt = int.Parse(Console.ReadLine());
        Console.Write("Caractér : " );
        String char = Console.ReadLine();
        copa(alt , char ) ;
    }

    //::::::::::::::::::::::::::::::::::::::::::::::::::
    //User Defined Functions Code
    //::::::::::::::::::::::::::::::::::::::::::::::::::
    public static void linhaCopa( int sp , int chars , String txt){
        for( int i = 0; i < sp; i = i + 1 ) {
            Console.Write(" " );
        }
        for( int i = 0; i < chars; i = i + 1 ) {
            Console.Write(txt );
        }
        Console.Write("\n" );
    }
    //::::::::::::::::::::::::::::::::::::::::::::::::::
    public static void copa( int alt , String txt){
        for( int i = 1; i < alt + 1; i = i + 1 ) {
            int sp = alt - i ;
            int chrs1 = i * 2 - 1 ;
            linhaCopa(sp , chrs1 , txt ) ;
        }
    }

}
//C#  - Automatic translated by Algorithmi 22.05
//(c) Ant0nio M@nso 	quarta, 2022 out 12 20:15
